tracer/build/_build/Build.VariableGenerations.cs

```diff
@@ -1234,11 +1235,63 @@ List<string> GetTracerStagesThatWillNotRun(string[] gitChanges)
        });
 
     static bool IsGitBaseBranch(string baseBranch)
-        => string.Equals(
+    {
+        // *****
+        // First we try to find the base branch in the Azure Pipelines environment variables.
+        // This code is a simplified version of Datadog.Trace/CI/CIEnvironmentValues.cs file
+        // to only extract the branch name from Azure Pipelines
+        // *****
+        const string AzureSystemPullRequestSourceBranch = "SYSTEM_PULLREQUEST_SOURCEBRANCH";
+        const string AzureBuildSourceBranch = "BUILD_SOURCEBRANCH";
+        const string AzureBuildSourceBranchName = "BUILD_SOURCEBRANCHNAME";
+
+        var prBranch = Environment.GetEnvironmentVariable(AzureSystemPullRequestSourceBranch);
+        var branch = !string.IsNullOrWhiteSpace(prBranch) ? prBranch : Environment.GetEnvironmentVariable(AzureBuildSourceBranch);
+        if (string.IsNullOrWhiteSpace(branch))
+        {
+            branch = Environment.GetEnvironmentVariable(AzureBuildSourceBranchName);
+        }
+        
+        if (string.Equals(CleanBranchName(branch), CleanBranchName(baseBranch), StringComparison.OrdinalIgnoreCase))
+        {
+            return true;
+        }
+
+        // *****
+        // In case we cannot find the baseBranch we fallback to the git command
+        // *****
+        return string.Equals(
             GitTasks.Git("rev-parse --abbrev-ref HEAD").First().Text,
             baseBranch,
             StringComparison.OrdinalIgnoreCase);
 
+        static string CleanBranchName(string branchName)
+        {
+            try
+            {
+                // Clean branch name
+                var regex = new Regex(@"^refs\/heads\/tags\/(.*)|refs\/heads\/(.*)|refs\/tags\/(.*)|refs\/(.*)|origin\/tags\/(.*)|origin\/(.*)$");
+                if (!string.IsNullOrEmpty(branchName))
+                {
+                    var match = regex.Match(branchName);
+                    if (match.Success && match.Groups.Count == 7)
+                    {
+                        branchName =
+                            !string.IsNullOrWhiteSpace(match.Groups[2].Value) ? match.Groups[2].Value :
+                            !string.IsNullOrWhiteSpace(match.Groups[4].Value) ? match.Groups[4].Value :
+                                                                                match.Groups[6].Value;
+                    }
+                }
+            }
+            catch (Exception ex)
```

Comment:
This is causing a warning:
```
The variable 'ex' is declared but never used
```

```suggestion
            catch
```